### PR TITLE
Update for MMTk core PR #306 - refactor TLS

### PR DESCRIPTION
**Pull request by qinsoon**

### CODE REVIEW

File: mmtk/src/active_plan.rs [hunk 1]
```diff
@@ -33,16 +33,18 @@ impl ActivePlan<JikesRVM> for VMActivePlan {
         &*SINGLETON.plan
     }
 
-    unsafe fn is_mutator(tls: OpaquePointer) -> bool {
-        let thread: Address = mem::transmute(tls);
-        !(thread + IS_COLLECTOR_FIELD_OFFSET).load::<bool>()
+    fn is_mutator(tls: VMThread) -> bool {
+        let thread: Address = unsafe { mem::transmute(tls) };
```
Comment (javadamiri) [inline mmtk/src/active_plan.rs:37]: Shouldn't we provide a safe interface for casting `VMThread` and `VMWorkerThread` to `Address`?
I doesn't look unsafe to me

**Review by javadamiri — APPROVED** [review]
LGTM

File: mmtk/src/active_plan.rs (hunk 1 above)
Comment (qinsoon) [reply mmtk/src/active_plan.rs:37]: This was intentional. In mmtk-core, we do not assume those are addresses, and we should not dereference them in any way. See here: https://github.com/mmtk/mmtk-core/blob/45c6a9b0c60d2888811a82c823b204cd20f60ab3/src/util/opaque_pointer.rs#L5

The question is whether we should provide a function for bindings to cast them. I feel the binding could implement their own functions to create and dereference the opaque pointers.

Comment (javadamiri) [reply mmtk/src/active_plan.rs:37]: I see

**Pull request merged**